tool/readme_scripts/readme_scripts_runner.dart

```diff
@@ -0,0 +1,184 @@
+  if (args.isNotEmpty) {
+    dirPath = args[0];
+    if (args.length == 2) {
+      category = args[1];
+    } else {
+      category = '';
+    }
+    final directory = Directory(dirPath);
+    if (directory.existsSync()) {
+      filePaths = getFilePaths(directory);
+      if (filePaths.isEmpty) {
+        print('No files identified, please check the provided path.');
+        exit(1);
+      }
+    } else {
+      print('The directory does not exist: $dirPath.');
+      exit(1);
+    }
+  } else {
+    print('Invalid arguments, please provide the path to a sample directory.');
+    exit(1);
+  }
+
+  var returnCode = 0;
+
+  for (final filePath in filePaths) {
+    final pathParts = filePath.split(Platform.pathSeparator);
+    final filename = pathParts.last;
+    final lFilename = filename.toLowerCase();
+
+    // skip any files that aren't the readme or metadata files
+    if (lFilename != lReadme && lFilename != lMetadata) {
+      continue;
+    }
+
+    if (lFilename == lReadme) {
+      // exit if readme filename capitalization is incorrect
+      if (filename != 'README.md') {
+        print(
+            'Error: readme file has wrong capitalization in filename. Should be: README.md');
+        returnCode++;
+      }
+      // get the readme file
+      readmeFile = filename;
+    }
+
+    // get the metadata file
+    if (lFilename == lMetadata) {
+      // exit if metadata filename capitalization is incorrect
+      if (filename != 'README.metadata.json') {
+        print(
+            'Error: metadata file has wrong capitalization in filename. Should be: README.metadata.json');
+        returnCode++;
+      }
+      metadataFile = filePath;
+    }
+  }
+
+  // check the readme exists
+  if (readmeFile == null) {
+    print(
+        'Error: README.md does not exist. Please create and re-run the script.');
+    returnCode++;
+  }
+
+  // Run readme checks
+  // check filename capitalization
+  // run the markdown linter style checker on the readme
+  returnCode += runMdlStyleChecker(dirPath);
+
+  // run the readme content check
+  returnCode += runReadmeCheck(dirPath);
+
+  // if the metadata file doesn't exist, create it from the readme
+  if (metadataFile == null) {
+    returnCode += createMetadataFromReadme(dirPath, category);
+    // run the metadata content check
+    returnCode += runMetadataCheck(dirPath);
+  } else {
+    // run the metadata content check
+    returnCode += runMetadataCheck(dirPath);
+  }
+
+  if (returnCode != 0) {
+    exit(returnCode);
+  } else {
+    exit(0);
+  }
+}
+
+int createMetadataFromReadme(String dirPath, String category) {
+  print('**** Creating README.metadata.json from README.md ****');
+  var result = Process.runSync('python3',
+      ['create_metadata_from_README.py', '-s', dirPath, '-c', category]);
+  printScriptOutput('create readme from metadata script', result);
+  return result.exitCode;
+}
+
+List<String> getFilePaths(Directory directory) {
+  final files = <String>[];
```

Comment:
This could all be a single statement `=> directory.listSync().where((e) => e is File).map((f) => f.path);` or something, but maybe that's overkill.

Reply:
Dart learning of the day - there's also a `whereType<Type>()` method which makes it even more readable.